external_search.py

```diff
@@ -132,8 +138,12 @@ def set_works_index_and_alias(self, current_alias):
         """Finds or creates the works_index and works_alias based on
         provided configuration.
         """
-        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
-        found = bool(index_details) and not (index_details.get('status')==404 or 'error' in index_details)
```

Comment:
If current_alias is None, self.indices.get_alias returns all the indices that have no alias. That makes index_details some arbitrary index that you probably don't want to use. I have lots of those in my local elasticsearch, and it could happen in production if you have an old index that you haven't deleted yet.